75 update project details

schultztimothy opened the pull request:
Fixing bugs mentioned here: https://github.com/gitcoinco/grants-hub/issues/75

gravityblast commented:
@schultztimothy  I remember we noticed we need only the creation date right? shall we change it to just check the timestamp of the ProjectCreated event?

gravityblast commented on client/src/components/grants/Show.tsx:
```diff
@@ -35,6 +42,34 @@ function Project() {
+        if (currentProject) {
+          const blockData = await global.web3Provider?.getBlock(
+            currentProject.block
+          );
```
I think we can get the block timestamp in the actions just after retrieving the project so that we have all the logic there and not in the UI,  what do you think?

gravityblast replied:
when we get the `ProjectCreated` events, the result already has a `getBlock` function

schultztimothy replied:
My thought was that by fetching the timestamp on display we won't need to make (project * n) requests before displaying the list of projects

gravityblast replied:
we could actually do 1 query filtering by an array of project ids

schultztimothy commented:
> @schultztimothy I remember we noticed we need only the creation date right? shall we change it to just check the timestamp of the ProjectCreated event?

We talked about not being sure which date should display but I don't think we got an answer. 

@michellema1208 I set this up to show the created at date unless the project was update, if the project was updated at some point it will show the updated at date. So it should always show the last date the project was created/updated. Just let us know what date should be displayed!

<img width="1015" alt="Screen Shot 2022-06-22 at 8 03 21 AM" src="https://user-images.githubusercontent.com/6887938/175050280-ecaac5fb-cc20-4b0e-b062-0493e1907b8b.png">

michellema1208 commented:
Hi @schultztimothy @gravityblast let's keep the current implementation! Agreed that it's most useful if it reflects last update. Can we add some text that says "last edited <date>" so it's clear to the user? @willsputra what do you think?

willsputra commented:
> Hi @schultztimothy @gravityblast let's keep the current implementation! Agreed that it's most useful if it reflects last update. Can we add some text that says "last edited " so it's clear to the user? @willsputra what do you think?

agree! we can show last updated date. maybe the text could be "last updated"?

schultztimothy commented:
> > Hi @schultztimothy @gravityblast let's keep the current implementation! Agreed that it's most useful if it reflects last update. Can we add some text that says "last edited " so it's clear to the user? @willsputra what do you think?
> 
> agree! we can show last updated date. maybe the text could be "last updated"?

@michellema1208 @willsputra one more clarification 😄 . @gravityblast updated the event that is fired when a project is updated, and that event no longer emits the address of who updated the project. This was done for the pinning service and makes it so that you are no longer able to query the projects a user has updated. Its definitely still possible to query this info and will be easier to do so when we implement the subgraph

Since the updated at timestamp seems more like a nice to have, would you all be ok with putting off displaying the updated at timestamp until we implement the subgraph?

gravityblast commented:
@schultztimothy we can query with a list of project ids, wdyt? I added a commit here that works in this branch:

https://github.com/gitcoinco/grants-hub/tree/filter-by-project-ids

but it's based on your previous commits where you had the `aggregateEvents` function and I can't open a PR that merges to this PR's branch.

schultztimothy commented:
> @schultztimothy we can query with a list of project ids, wdyt? I added a commit here that works in this branch:
> 
> https://github.com/gitcoinco/grants-hub/commits/filter-by-project-ids
> 
> but it's based on your previous commits where you had the `aggregateEvents` function and I can't open a PR that merges to this PR's branch.

Thanks for putting that together! I added it in to this branch
